src/redux/reducers/compare.ts

```diff
@@ -0,0 +1,385 @@
+      }
+    },
+    resourceSetFetched: (state, action: PayloadAction<{side: CompareSide; resources: K8sResource[]}>) => {
+      const {side, resources} = action.payload;
+      state.current[side] = {
+        error: false,
+        loading: false,
+        resources,
+      };
+
+      const leftReady = state.current.left && !state.current.left.loading && !state.current.left.error;
+      const rightReady = state.current.right && !state.current.right.loading && !state.current.right.error;
+      if (leftReady && rightReady) {
+        state.current.diff = {
+          loading: true,
+          comparisons: [],
+        };
+      }
+    },
+    resourceSetFetchFailed: (state, action: PayloadAction<{side: CompareSide; reason: string}>) => {
+      const {side, reason} = action.payload;
+      log.debug('[compare] resource set fetch failed', reason);
+
+      state.current[side] = {
+        error: true,
+        loading: false,
+        resources: [],
+      };
+    },
+    resourceSetCompared: (state, action: PayloadAction<{comparisons: ResourceComparison[]}>) => {
+      state.current.diff = {
+        loading: false,
+        comparisons: action.payload.comparisons,
+      };
+    },
+  },
+});
+
+/* * * * * * * * * * * * * *
+ * Export
+ * * * * * * * * * * * * * */
+export default compareSlice.reducer;
+export const {
+  compareToggled,
+  filterUpdated,
+  operationUpdated,
+  resourceSetCleared,
+  resourceSetRefreshed,
+  resourceSetSelected,
+  resourceSetFetchFailed,
+  resourceSetFetched,
+  resourceSetCompared,
+  diffViewOpened,
+  comparisonAllToggled,
+  comparisonToggled,
+} = compareSlice.actions;
+
+/* * * * * * * * * * * * * *
+ * Selectors
+ * * * * * * * * * * * * * */
+export type CompareStatus = 'selecting' | 'comparing';
+export const selectCompareStatus = (state: CompareState): CompareStatus => {
+  const c = state.current;
+
+  const empty = !c.left && !c.right;
+  const leftSuccess = c.left && !c.left.loading && !c.left.error;
+  const rightSuccess = c.right && !c.right.loading && !c.right.error;
+  if (empty || !leftSuccess || !rightSuccess) {
+    return 'selecting';
+  }
+
+  return 'comparing';
+};
+
+export const selectDiffedComparison = (state: CompareState): MatchingResourceComparison | undefined => {
+  const id = state.current.viewDiff;
+  if (!id) return undefined;
+  const comparison = state.current.diff?.comparisons.find(c => c.id === id);
+  if (!comparison || !comparison.isMatch) return undefined;
+  return comparison;
+};
+
+export const selectIsComparisonSelected = (state: CompareState, id: string): boolean => {
+  return state.current.selection.some(comparisonId => comparisonId === id);
+};
+
+export const selectIsAllComparisonSelected = (state: CompareState): boolean => {
+  return !state.current.diff?.loading && state.current.selection.length === state.current.diff?.comparisons.length;
+};
+
+/* * * * * * * * * * * * * *
+ * Listeners
+ * * * * * * * * * * * * * */
+startAppListening({
```

Comment:
This registration should be done somewhere else and simply contain definition here. Loading this file should not have this side-effect. There are two disadvantages with current approach:
1. Does not allow using mocked listeners.
2. Does not have a name that identifies it, this would need a comment to give it a meaningful name.